Cargo.toml

```diff
@@ -33,6 +34,9 @@ esp-backtrace = { git = "https://github.com/bjoernQ/esp-backtrace.git", features
 esp-println = { git = "https://github.com/esp-rs/esp-println.git", features = [ "esp32c3" ] }
```

Comment:
esp32c3 is riscv32imc, so we should probably change all references of riscv32i to riscv32imc. Alternatively, we could cfg by target arch?

Reply:
mhhh it just didn't work for me with anything bit "riscv32i" - need to check that again

Reply:
Just to double check, I stashed my local changes and build for `riscv32i`, unfortunately still getting the linker errors.

Reply:
it _seems_ to work for me also with `imc` ... no idea why it didn't in the beginning 🤷‍♂️ 

and I can still compile it after a full clean and `cargo update` etc.

I am on `rustc 1.62.0-nightly (60e50fc1c 2022-04-04)` currently .... will update and see if it breaks then

Reply:
after updating I'm on `rustc 1.63.0-nightly (42e1761c7 2022-05-15)` and still everything works

Reply:
Ah ha! Updating nightly fixed it.